src/ert/plugins/plugin_manager.py

```diff
@@ -139,6 +140,50 @@ def get_flow_config_path(self) -> Optional[str]:
             hook=self.hook.flow_config_path, config_name="flow"
         )
 
+    def get_forward_model_configuration(self) -> Dict[str, Dict[str, Any]]:
+        response: List[PluginResponse[Dict[str, str]]] = (
+            self.hook.forward_model_configuration()
+        )
+        if response == []:
+            return {}
+
+        fm_configs: Dict[str, Dict[str, Any]] = collections.defaultdict(dict)
+        for res in response:
+            if not isinstance(res.data, dict):
+                raise TypeError(
+                    f"{res.plugin_metadata.plugin_name} did not provide "
+                    "dict[str, dict]"
+                )
+
+            for fmstep_name, fmstep_config in res.data.items():
+                if not isinstance(fmstep_name, str):
+                    raise TypeError(
+                        f"{res.plugin_metadata.plugin_name} did not "
+                        "provide dict[str, dict[str, Any]]"
+                    )
+                if not isinstance(fmstep_config, dict):
+                    raise TypeError(
+                        f"{res.plugin_metadata.plugin_name} did not "
+                        "provide dict[str, dict[str, Any]]"
+                    )
```

Comment:
I think this should either be joined in one if statement or provide a more accurate error message for each case :thinking:

Reply:
fixed